From: Theodore Ts'o <tytso@mit.edu>
Date: Fri, 20 Apr 2018 10:15:17 -0400
X-Dgit-Generated: 2.08-2 3a6150c82b8fb418b761bf3d46983ed4730746fe
Subject: Fix unreliable support of --capitalize when --ambiguous is requested

When generating passwords using the default phonemes mechanism,
--capitalize isn't reliably supported when ambiguous characters are
removed.  Fix this.

Signed-off-by: Theodore Ts'o <tytso@mit.edu>

---

--- pwgen-2.08.orig/pw_phonemes.c
+++ pwgen-2.08/pw_phonemes.c
@@ -111,7 +111,7 @@ try_again:
 			buf[c+len] = '\0'; /* To make strpbrk() happy */
 			cp = strpbrk(buf, pw_ambiguous);
 			if (cp)
-				continue;
+				goto try_again;
 		}
 		
 		c += len;
